packages/angular/ssr/src/routes/route-tree.ts

```diff
@@ -49,13 +50,30 @@ export interface RouteTreeNodeMetadata {
    * structure and content of the application.
    */
   route: string;
+
+  /**
+   * Optional status code to return for this route.
+   */
+  status?: number;
+
+  /**
+   * Optional additional headers to include in the response for this route.
+   */
+  headers?: Record<string, string>;
+
+  /**
+   * Specifies the rendering mode used for this route.
+   * If not provided, the default rendering mode for the application will be used.
+   */
+  renderMode?: RenderMode;
 }
 
 /**
  * Represents a node within the route tree structure.
  * Each node corresponds to a route segment and may have associated metadata and child nodes.
+ * The `additionalMetadata` type parameter allows for extending the node metadata with custom data.
  */
-interface RouteTreeNode {
+interface RouteTreeNode<additionalMetadata> {
```

Comment:
Suggestion: Should we do `AdditionalMetadata extends {}` (or `Record<string, unknown>`)? We probably wouldn't want a string or boolean here.